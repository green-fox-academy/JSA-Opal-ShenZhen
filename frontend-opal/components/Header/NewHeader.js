import React, { Component } from 'react';
import { Container, Header, Item, Input, Icon, Left, Body, Right, Button, Title, Text } from 'native-base';

const NewHeader = () => {
    return (
        <Container>
            <Header searchBar rounded>
                <Icon name="search"></Icon>
                <Item>
                    <Icon name="ios-search" />
                    <Input placeholder="Search" />
                    <Icon name="ios-people" />
                </Item>

                <Button transparent>
                    <Text>Search</Text>
                </Button>
            </Header>
        </Container>
    )
}

export default NewHeader;