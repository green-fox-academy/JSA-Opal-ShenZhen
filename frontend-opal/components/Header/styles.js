import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    search: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-around",
        width: "100%",
        height: 50,
        alignContent: "center",
        backgroundColor: "#0984e3"
    },
    listIcon: {
        marginTop: 14,
        marginLeft: 20,
        width: 25,
        height: 25,

    },
    input: {
        marginTop: 10,
        width: "75%",
        height: 30,
        backgroundColor: "white",
    },
    searchButton: {
        marginTop: 14,
        marginRight: 20,
        width: 25,
        height: 25,
    },

    navBars: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "space-between",

    },
    narTitle: {
        padding: 2,
        width: "50%",
        height: 50,
        textAlign: "center",
        borderBottomWidth: 1,
        borderBottomColor: "#e17055",
        fontSize: 12
    },
})