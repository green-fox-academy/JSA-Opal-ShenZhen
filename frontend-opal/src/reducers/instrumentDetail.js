const pairs = {
  GET_INSTRUMENT_DETAIL_DATA: (state, action) => action.payload.data,
  CLEAR_DETAIL_DATA: (state, action) => []
};

export default pairs;
