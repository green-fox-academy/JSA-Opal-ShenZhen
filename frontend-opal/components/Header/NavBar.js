import React from 'react';
import { Button, View } from 'react-native';
import styles from './styles';

const NavBar = ({ navTitles }) => {
  const handlePress = key => key === 'PORTFOLIO';

  const renderItems = titles =>
    titles.map(title => (
      <View
        key={title.id}
        style={[styles.navTitle, handlePress(title.key) && styles.buttonPressed]}
      >
        <Button title={title.key} color="#e17055" />
      </View>
    ));

  return <View style={styles.navBars}>{renderItems(navTitles)}</View>;
};

export default NavBar;
