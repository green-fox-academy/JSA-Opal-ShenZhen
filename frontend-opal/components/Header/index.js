import React from 'react'
import {
    Button,
    View,
    TextInput,
    Image,
} from 'react-native'
import styles from './styles'
import ListIcon from '../../assets/list-icon.png';
import SearchIcon from '../../assets/search-button.png';

const Header = () => {
    const narTitles = [
        { id: 1, key: "PORTFOLI" },
        { id: 2, key: "WATCHLIST" }
    ]

    const handlePress = (key) => key === "PORTFOLIO";

    const renderItems = (titles) => (
        titles.map(title => (
          <View
            key={title.id}
            style={[styles.narTitle, handlePress(title.key) && styles.buttonPressed]}
          >
            <Button
              title={title.key}
              color="#e17055"
            />
          </View>
        ))
    )

    return (
      <View>
        <View style={styles.search}>
          <Image
            style={styles.listIcon}
            source={ListIcon}
          />
          <TextInput style={styles.input} />
          <Image
            style={styles.searchButton}
            source={SearchIcon}
          />
        </View>

        <View style={styles.navBars}>
          {renderItems(narTitles)}
        </View>
      </View>
    );
}

export default Header
