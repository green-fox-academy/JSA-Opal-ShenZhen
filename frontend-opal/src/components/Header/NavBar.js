import React from 'react';
import PropTypes from 'prop-types';
import { Button, View } from 'react-native';
import styles from './styles';

const NavBar = ({ navTitles }) => {
  const handlePress = (key) => key === 'PORTFOLIO';

  const renderItems = (titles) => titles.map((title) => (
    <View
      key={title.id}
      style={[styles.navTitle, handlePress(title.key) && styles.buttonPressed]}
    >
      <Button title={title.key} color="#e17055" />
    </View>
  ));

  return <View style={styles.navBars}>{renderItems(navTitles)}</View>;
};

NavBar.propTypes = {
  navTitles: PropTypes.objectOf(
    PropTypes.shape({
      id: PropTypes.number,
      key: PropTypes.string,
    }),
  ).isRequired,
};

export default NavBar;
