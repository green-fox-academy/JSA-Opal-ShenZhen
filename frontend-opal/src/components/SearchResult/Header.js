import React from 'react';
import { Header, Button, Icon, Left, Body, Right, Input, Item } from 'native-base';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import thunks from 'thunks/searchResult';
import styles from './styles/header';

function SearchHeader({ search }) {
  return (
    <Header style={styles.container}>
      <Left>
        <Button transparent>
          <Icon name="arrow-back" />
        </Button>
      </Left>
      <Body>
        <Item searchBar style={styles.searchContainer}>
          <Icon name="search" style={{ color: 'white' }} />
          <Input
            placeholder="Search"
            style={{ color: 'white' }}
            onChange={e => search(e.nativeEvent.text)}
          />
        </Item>
      </Body>
      <Right />
    </Header>
  );
}

SearchHeader.propTypes = {
  search: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  search: input => dispatch(thunks.search(input))
});

export default connect(null, mapDispatchToProps)(SearchHeader);
