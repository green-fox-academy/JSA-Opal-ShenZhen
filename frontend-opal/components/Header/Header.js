import React from 'react';
import { View, TextInput, Image } from 'react-native';
import styles from './styles';
import ListIcon from '../../assets/list-icon.png';
import SearchIcon from '../../assets/search-button.png';

import NavBar from './NavBar';

const Header = () => {
  const navTitles = [
    { id: 1, key: 'PORTFOLIO' },
    { id: 2, key: 'WATCHLIST' }
  ];

  return (
    <View>
      <View style={styles.search}>
        <Image style={styles.listIcon} source={ListIcon} />
        <TextInput style={styles.input} />
        <Image style={styles.searchButton} source={SearchIcon} />
      </View>

      <NavBar navTitles={navTitles} />
    </View>
  );
};

export default Header;
