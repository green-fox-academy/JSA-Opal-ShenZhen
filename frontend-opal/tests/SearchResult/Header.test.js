import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducers from 'reducers/root';
import Header from 'components/SearchResult/Header';

describe('<Header />', () => {
  it('renders correctly', () => {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
    const tree = renderer.create(
      <Provider store={store}>
        <Header />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
