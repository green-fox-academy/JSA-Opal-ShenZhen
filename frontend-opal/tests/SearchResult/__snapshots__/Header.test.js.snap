// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        },
        Object {
          "backgroundColor": "rgb(77,119,245)",
          "elevation": 20,
          "marginTop": 58,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": -3,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 4,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 4,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 0,
            "borderRadius": undefined,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "borderWidth": undefined,
            "flexDirection": "row",
            "height": 60,
            "marginLeft": 2,
            "opacity": 1,
            "width": 480,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChange={[Function]}
          placeholder="Search"
          placeholderTextColor="#575757"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 5,
                "paddingRight": 5,
                "top": 1.5,
              },
              Object {
                "color": "white",
              },
            ]
          }
          underlineColorAndroid="rgba(0,0,0,0)"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    />
  </View>
</View>
`;
